src/nimdowpkg/config/config.nim

```diff
@@ -0,0 +1,85 @@
+import 
+  x11 / [x,  xlib],
+  parsetoml,
+  tables,
+  "../keys/keyutils",
+  "../event/xeventmanager",
+  "../windowmanger"
+
+# Mapping of config keys to functions.
+const ProcTable: Table[string, proc] = {
+  "testAction": testAction,
+  "testAction2": testAction2
+}.toTable
```

Comment:
I dislike the config module having so many dependencies, but especially dependencies on window manager related modules. We can likely get rid of the `windowmanager` dependency by providing some sort of interface or map to `config.nim`

Reply:
Fixed this by adding a function which allows other modules to map `action` names to procs. The actions must be added **before** the config file is loaded, or an error will be thrown.